### PR TITLE
Support custom rollup resolution

### PR DESCRIPTION
**Pull request by cody271**

### CODE REVIEW

File: src/datasource.ts [hunk 1]
```diff
@@ -6,6 +6,29 @@ import { metaInterpolateLabel } from './irondb_query';
...
+const DURATION_UNITS = {
+  ms: 1,
+  s: 1000,
+  m: 1000 * 60,
+  h: 1000 * 60 * 60,
+  d: 1000 * 60 * 60 * 24,
+};
+
+function parseDurationMs(duration: string): number {
+  if (!/^[0-9]+(ms|s|m|h|d)?$/g.test(duration)) {
```
Comment (postwait) [inline src/datasource.ts]: I would lowercase the duration string before the match.

File: src/datasource.ts [hunk 2]
```diff
@@ -404,34 +428,62 @@ export default class IrondbDatasource {
...
+    if (rolluptype !== 'automatic' && _.isEmpty(metricrollup)) {
+      rolluptype = 'automatic';
+      log(() => `getRollupSpan() defaulting to automatic`);
+    }
+    let rollupMs;
+    if (rolluptype === 'exact') {
+      const exactMs = parseDurationMs(metricrollup);
+      const exactDatapoints = Math.floor(((end - start) * 1000) / exactMs);
+      log(() => `getRollupSpan() exactMs = ${exactMs}, exactDatapoints = ${exactDatapoints}`);
+      if (exactDatapoints * IrondbDatasource.MAX_DATAPOINTS_THRESHOLD > options.maxDataPoints) {
```
Comment (postwait) [inline src/datasource.ts]: Let's make MAX_DATAPOINTS_THRESHOLD and MAX_EXACT_DATAPOINTS_THRESHOLD two separate constants that both happen to be 1.5 right now.  I have a feeling people are going to say: "I said 5ms rollups, stop telling me I can't" so we might need to up this threshold here, but there's no reason to up the other case.

Comment (postwait) [reply src/datasource.ts]: I think you multiplied on the wrong side of this. this should be:
`exactDatapoints > options.maxDataPoints * IrondbDatasource.MAX_EXACT_DATAPOINTS_THRESHOLD`

File: src/datasource.ts [hunk 3]
```diff
@@ -287,30 +310,28 @@ export default class IrondbDatasource {
...
-      data['period'] = interval;
-      data['start'] = start;
-      data['count'] = Math.round((end - start) / interval);
-      data['reduce'] = [{ label: '', method: reduce }];
       for (let i = 0; i < irondbOptions['std']['names'].length; i++) {
+        options = {};
+        options.url = this.url;
+        if ('hosted' === this.irondbType) {
+          options.url = options.url + '/irondb';
+          options.url = options.url + '/fetch';
```
Comment (bdunavant) [inline src/datasource.ts]: This works fine, but should really be a single operation, with an else into the standalone version to bring those closer together.  nit.

**Pull request merged**